data/dataDefaulter.js

```diff
@@ -0,0 +1,51 @@
+function addDefaultsToResults(
+    scrapedResults,
+    cachedResults,
+    secondsOfTolerance
+) {
+    if (!(cachedResults && cachedResults.length)) {
+        return scrapedResults;
+    } else {
+        const combinedResults = [];
+        const scrapeGrouping = {};
+        scrapedResults.forEach((result) => {
+            combinedResults.push(result);
+            scrapeGrouping[generateKey(result)] = 1;
+        });
+
+        cachedResults.forEach((cachedResult) => {
+            if (!scrapeGrouping[generateKey(cachedResult)]) {
+                if (secondsOfTolerance) {
+                    const lowerTimeBound =
+                        new Date() - secondsOfTolerance * 1000;
+                    if (
+                        cachedResult.timestamp &&
+                        cachedResult.timestamp >= lowerTimeBound
+                    ) {
+                        combinedResults.push(cachedResult);
+                    }
+                } else {
+                    combinedResults.push(cachedResult);
+                }
+            }
+        });
+
+        return combinedResults;
+    }
+}
+
+function generateKey(entry) {
+    let uniqueIdentifier = "";
+    ["name", "street", "city", "zip"].forEach((key) => {
+        if (entry[key]) {
+            uniqueIdentifier += `${entry[key]};`;
```

Comment:
How about adding some sort of separator like '|' so that we don't get overlaps however unlikely?